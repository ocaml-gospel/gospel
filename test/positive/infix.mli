val (==) : 'a -> 'a -> bool
(*@ r = (==) x y
      ensures r <-> x = y *)
