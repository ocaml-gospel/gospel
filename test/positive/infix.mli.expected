
*******************************
********** Parsed file ********
*******************************
val (==) : 'a -> 'a -> bool[@@gospel
                             {| r = (==) x y
      ensures r <-> x = y |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

val (==) : 'a -> 'a -> bool
(*@ r = infix == x y
    ensures ...
     *)

*******************************
********* Typed GOSPEL ********
*******************************
module infix.mli

  Namespace: infix.mli
    Type symbols
      
    Logic Symbols
      
    Exception Symbols
      
    Namespaces
      
    Type Namespaces
      
  Signatures
    (*@ open Gospelstdlib *)
    
    val == : 'a -> 'a -> bool
    (*@ r:bool = == x:'a y:'a
        ensures (r:bool = (True ):bool):prop <-> (x:'a = y:'a):prop*)

OK
